src/transport.js

```diff
@@ -53,11 +54,18 @@ module.exports = function (swarm) {
       // specific for the transport we are using
       const proxyConn = new Connection()
 
-      next(multiaddrs.shift())
+      each(multiaddrs, next)
```

Comment:
@diasdavid @dignifiedquire +1 to merging this as is for now. 

I've created an issue to track the enhancement proposed by @dignifiedquire